### PR TITLE
PAYEE-536: fix backlink issue for Update your estimated taxable incom…

### PR DESCRIPTION
**Pull request by sergi-hmrc**
…e journey

- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/views/incomes/EditSuccessView.scala.html [hunk 1]
```diff
@@ -32,8 +32,7 @@
     title = messages("tai.incomes.updated.check.title"),
     authedUser = Some(user),
     pageTitle = Some(Messages("tai.service.navTitle")),
-    backLinkContent = Some(Messages("tai.back-link.upper")),
-    backLinkId = "backLink",
+    backLinkContent = None,
```
Comment (PerrettJ4) [inline app/views/incomes/EditSuccessView.scala.html:35]: why is backLinkContent None, out of interest?

**Review by PerrettJ4 — APPROVED** [review]
Looks good to me :D
Probably best to get a second opinion tho 👍

File: app/views/incomes/BonusPaymentsView.scala.html [hunk 2]
```diff
@@ -28,15 +28,15 @@
...
+@(yesNoForm: Form[YesNoForm], employer: IncomeSource, backUrl: String)(implicit request: Request[_], messages: Messages, user: controllers.auth.AuthedUser,
 templateRenderer: uk.gov.hmrc.renderer.TemplateRenderer,  ec: ExecutionContext)
 
 @main(
     title = messages("tai.bonusPayments.title", TaxYearRangeUtil.currentTaxYearRangeBetweenDelimited),
     pageTitle = Some(messages("tai.service.navTitle")),
     authedUser = Some(user),
     backLinkContent = Some(Messages("tai.back-link.upper")),
-    backLinkUrl=Some(controllers.income.estimatedPay.update.routes.IncomeUpdatePayslipAmountController.taxablePayslipAmountPage.url)
+    backLinkUrl = Some(backUrl)
```
Comment (brendanstraughan) [inline app/views/incomes/BonusPaymentsView.scala.html:39]: should we add backLinkId in as well to keep it consistent?

File: test/controllers/income/estimatedPay/update/IncomeUpdateBonusControllerSpec.scala [hunk 3]
```diff
@@ -64,6 +64,8 @@ class IncomeUpdateBonusControllerSpec
       .thenReturn(Future.successful(Right(employer.id)))
     when(journeyCacheService.mandatoryJourneyValue(Matchers.eq(UpdateIncome_NameKey))(any()))
       .thenReturn(Future.successful(Right(employer.name)))
+    when(journeyCacheService.currentValue(Matchers.eq(UpdateIncome_TaxablePayKey))(any()))
+      .thenReturn(Future.successful(Some("taxablepay")))
```
Comment (brendanstraughan) [inline test/controllers/income/estimatedPay/update/IncomeUpdateBonusControllerSpec.scala:68]: Should we add a test for when this is None that the url is 
controllers.income.estimatedPay.update.routes.IncomeUpdatePayslipAmountController.payslipDeductionsPage().url

File: test/views/html/incomes/nextYear/UpdateIncomeCYPlus1ConfirmViewSpec.scala [hunk 4]
```diff
@@ -30,7 +30,13 @@ class UpdateIncomeCYPlus1ConfirmViewSpec extends TaiViewSpec {
...
   val employmentId = 1
 
-  val vm = ConfirmAmountEnteredViewModel(employmentId, employerName, currentAmount, estimatedAmount, IrregularPay)
+  val vm = ConfirmAmountEnteredViewModel(
+    employmentId,
+    employerName,
+    currentAmount,
+    estimatedAmount,
+    IrregularPay,
+    "javascript:history.go(-1)")
```
[the remark this answers is not in the document]
Comment (brendanstraughan) [reply test/views/html/incomes/nextYear/UpdateIncomeCYPlus1ConfirmViewSpec.scala:39]: ignore, this is because of the TODO, my bad

File: test/controllers/income/UpdateIncomeNextYearControllerSpec.scala [hunk 5]
```diff
@@ -442,7 +442,13 @@ class UpdateIncomeNextYearControllerSpec
...
           )
 
-          val vm = ConfirmAmountEnteredViewModel(employmentID, employerName, currentAmount, newAmount, NextYearPay)
+          val vm = ConfirmAmountEnteredViewModel(
+            employmentID,
+            employerName,
+            currentAmount,
+            newAmount,
+            NextYearPay,
+            "javascript:history.go(-1)")
```
[the remark this answers is not in the document]
Comment (brendanstraughan) [reply test/controllers/income/UpdateIncomeNextYearControllerSpec.scala:451]: ignore, because of the TODO

File: app/views/incomes/EditSuccessView.scala.html (hunk 1 above)
Comment (sergi-hmrc) [reply app/views/incomes/EditSuccessView.scala.html:35]: So the Success View in other views has never had the back link. I looked at it and it was introduced recently but I don't think it should be there

File: test/controllers/income/estimatedPay/update/IncomeUpdateBonusControllerSpec.scala (hunk 3 above)
Comment (sergi-hmrc) [reply test/controllers/income/estimatedPay/update/IncomeUpdateBonusControllerSpec.scala:68]: Yup, will do now :)

File: app/views/incomes/EditSuccessView.scala.html (hunk 1 above)
Comment (PerrettJ4) [reply app/views/incomes/EditSuccessView.scala.html:35]: Thank you!
